File: packages/js-sdk/src/types/quote-template.ts
/**
 * TypeScript types for TurboQuote — Quote Template entity and request/response types
 */

// ============================================
// DOMAIN TYPES
// ============================================

export interface QuoteTemplate {
  id: string;
  orgId: string;
  logoUrl: string | null;
  primaryColor: string;
  primaryTextColor: string;
  disclaimer: string | null;
  termsAndConditions: string | null;
  closingMessage: string | null;
  senderName: string | null;
  senderPhone: string | null;
  senderEmail: string | null;
  contactEmail: string | null;
  isActive: boolean;
  createdBy: string | null;
  createdOn: string;
  updatedOn: string;
}

// ============================================
// REQUEST TYPES
// ============================================

export interface CreateQuoteTemplateRequest {
  logoUrl?: string;
  primaryColor?: string;
  primaryTextColor?: string;
  disclaimer?: string;
  termsAndConditions?: string;
  closingMessage?: string;
  senderName?: string;
  senderPhone?: string;
  senderEmail?: string;
  contactEmail?: string;
}

export type UpdateQuoteTemplateRequest = CreateQuoteTemplateRequest;

File: packages/py-sdk/src/turbodocx_sdk/types/quote_template.py
"""
TypedDict definitions for TurboQuote -- Quote Template entity.
"""

from __future__ import annotations

from typing import Any, Dict, List, Optional
from typing_extensions import TypedDict


class QuoteTemplate(TypedDict, total=False):
    id: str
    orgId: str
    logoUrl: Optional[str]
    primaryColor: str
    primaryTextColor: str
    disclaimer: Optional[str]
    termsAndConditions: Optional[str]
    closingMessage: Optional[str]
    senderName: Optional[str]
    senderPhone: Optional[str]
    senderEmail: Optional[str]
    contactEmail: Optional[str]
    isActive: bool
    createdBy: Optional[str]
    createdOn: str
    updatedOn: str


class CreateQuoteTemplateRequest(TypedDict, total=False):
    logoUrl: str
    primaryColor: str
    primaryTextColor: str
    disclaimer: str
    termsAndConditions: str
    closingMessage: str
    senderName: str
    senderPhone: str
    senderEmail: str
    contactEmail: str


# Same shape as create
UpdateQuoteTemplateRequest = CreateQuoteTemplateRequest


class QuoteTemplateListResponse(TypedDict):
    results: List[QuoteTemplate]
    totalRecords: int
